sqlglot/dialects/presto.py

```diff
@@ -149,6 +149,11 @@ def _sequence_sql(self: generator.Generator, expression: exp.GenerateSeries) ->
 
     sql = self.func("SEQUENCE", start, end, step)
     if isinstance(expression.parent, exp.Table):
+        if expression.parent.alias:
```

Comment:
i don't understand how this code works. first of all you're mutating expression.parent but you shouldn't be.

but second of all sql is already generated at this point, this line of code isn't really doing what you're intending to do

Reply:
Yeah, has a chat with George, working on a proper solution!

Reply:
Probably the reason it works is it generates the sql for the leaf node first  and then for the parent, an alias?